libcudacxx/include/cuda/std/__cccl/exceptions.h

```diff
@@ -32,4 +32,54 @@
 #  define _CCCL_HAS_EXCEPTIONS() __EXCEPTIONS
 #endif // has exceptions
 
+// The following macros are used to conditionally compile exception handling code. They
+// are used in the same way as `try` and `catch` blocks, but they allow for different
+// behavior based on whether exceptions are enabled or not, and whether the code is being
+// compiled for device or not.
+//
+// Usage:
+//   _CCCL_TRY({
+//     ...                             // Code that may throw an exception
+//   })
+//   _CCCL_CATCH_OR((cuda_error& e) {  // Handle CUDA exceptions
+//     ...
+//   })
+//   _CCCL_CATCH((...) {               // Handle any other exceptions
+//     ...
+//   })
+#if !_CCCL_HAS_EXCEPTIONS() || (_CCCL_DEVICE_COMPILATION() && !_CCCL_CUDA_COMPILER(NVHPC))
+#  define _CCCL_TRY(...) {__VA_ARGS__}
```

Comment:
I am not a fan of this taking the whole statement as an argument, could we rather have a `_CCCL_TRY()` macro that expands to just `try`?

Reply:
if nvc++ permitted this, it would solve the problem:
```c++
if target (...) {
  ...
}
else
  try {
    ...
  }
  catch(foo) {
    ...
  }
  catch(...) {
    ...
  }
```

but nvc++ rejects an `else` without curlies for `if target`.

Reply:
@ericniebler I believe we should simplify this with that knowledge